### PR TITLE
338 iroha registration

### PR DESCRIPTION
**Pull request by mrZizik**

### CODE REVIEW

File: iroha-android-sample/src/main/java/jp/co/soramitsu/iroha/android/sample/interactor/CreateAccountInteractor.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import static jp.co.soramitsu.iroha.android.sample.Constants.DOMAIN_ID;
+import static jp.co.soramitsu.iroha.android.sample.Constants.PRIV_KEY;
+import static jp.co.soramitsu.iroha.android.sample.Constants.PUB_KEY;
+import static jp.co.soramitsu.iroha.android.sample.Constants.TX_COUNTER;
+
+public class CreateAccountInteractor extends CompletableInteractor<String> {
+
+    private final ManagedChannel channel;
+    @Inject
+    ModelCrypto crypto;
```
Comment (BulatMukhutdinov) [inline iroha-android-sample/src/main/java/jp/co/soramitsu/iroha/android/sample/interactor/CreateAccountInteractor.java]: Why do you inject model cpypto? Does it needed?

File: iroha-android-sample/src/main/res/layout/activity_registration.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                <TextView
+                    android:id="@+id/subtitle"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center_horizontal"
+                    android:text="@string/registration_subtitle"
+                    android:textColor="@color/hint" />
+
+                <EditText
+                    android:id="@+id/username"
```
Comment (BulatMukhutdinov) [inline iroha-android-sample/src/main/res/layout/activity_registration.xml]: Add iroha username constraints

**Comment by BulatMukhutdinov** [comment]
Add timeout to registration. Waited for 5 minutes - progress window is still in progress

**Comment by BulatMukhutdinov** [comment]
Also, pls check if user already registered - redirect him to the main screen

File: iroha-android-sample/src/main/java/jp/co/soramitsu/iroha/android/sample/interactor/CreateAccountInteractor.java (hunk 1 above)
Comment (mrZizik) [reply iroha-android-sample/src/main/java/jp/co/soramitsu/iroha/android/sample/interactor/CreateAccountInteractor.java]: We need it to generate keys on 55 line

**Pull request merged**